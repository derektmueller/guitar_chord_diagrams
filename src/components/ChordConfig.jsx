import React from 'react';
import { connect } from 'react-redux'
import { Field, reduxForm } from 'redux-form'
import { addChordDiagram } from '../actions/chordDiagrams'

let ChordConfig = ({handleSubmit}) => {
  return (<form className='config' onSubmit={handleSubmit}>
    <Field name="title" component="input" type="text" />
    <Field name="notes" component="input" type="text" />
    <Field name="root" component="input" type="text" />
    <Field name="fretCount" component="input" type="text" />
    <Field name="tuning" component="input" type="text" />
    <button type="submit">Submit</button>
  </form>);
};

export default reduxForm({
  form: 'chord-config',
  onSubmit: (values, dispatch) => {
    dispatch(addChordDiagram(values));
    return false;
  }
})(ChordConfig);

