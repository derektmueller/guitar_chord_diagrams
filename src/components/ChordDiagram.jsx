import React from 'react';
import ChordDiagrams from '../../lib/ChordDiagrams.js';
import { removeChordDiagram } from '../actions/chordDiagrams'
import { connect } from 'react-redux'

export class ChordDiagram extends React.Component {
  fretWidth() {
    return 1 / (this.props.chordDiagrams.fretCount) * 100;
  }

  stringSpacing() {
    return 1 / (this.props.chordDiagrams.tuning.length) * 100;
  }

  renderFrets() {
    return (<div className='frets'>
      {(new Array(this.props.chordDiagrams.fretCount).fill(null)).map(
        (_, i) => {
          return <div className='fret' key={i} 
            style={{width: `${this.fretWidth()}%`}}></div>;
        })}
    </div>);
  }

  renderNoteLabel(note) {
    return note.split('').map((elem, i) => {
      if(i === 1) {
        switch(elem) {
          case 'b':
            return <sup key={i} className='flat' />;
          case '#':
            return <sup key={i} className='sharp' />;
        }
      } else {
        return <div key={i}>{elem}</div>;
      }
    });
  }

  renderNote(note, i) {
    return note ? 
      (<div 
        className={`note ${this.props.root === note ? ' root' : ''}`}
        key={`node-${i}`} 
        style={{
          left: `${i * this.fretWidth()}%`,
          background: `#${this.props.colorPalette.getColor(note)}`
        }}>{this.renderNoteLabel(note)}
      </div>) : null;
  }

  renderString(string, i) {
    return (
      <div style={{height: `${this.stringSpacing()}%`}} 
        className='string-container' key={i}>

        {[<div className='string' key={`string-${i}`}></div>]
          .concat(string.map(this.renderNote.bind(this)))}
      </div>);
  }

  renderStrings() {
    return this.props.chordDiagrams.getDiagram().map(
      this.renderString.bind(this));
  }

  renderDot(fret, i) {
    return <div className='dot'
      key={i}
      style={{
        left: 
          `${fret * this.fretWidth() - this.fretWidth() / 2}%`,
      }}>
      </div>;
  }

  renderDots() {
    return [3, 5, 7, 9, 15, 17]
      .filter((fret) => fret < this.props.chordDiagrams.fretCount)
      .map(this.renderDot.bind(this));
  }

  renderDoubleDots() {
    return <div className='double-dots'>
      {[12, 12]
        .filter((fret) => fret < this.props.chordDiagrams.fretCount)
        .map(this.renderDot.bind(this))}
      </div>;
  }

  renderDeleteButton() {
    return <div 
      className='delete' 
      onClick={() => 
        this.props.removeChordDiagram(this.props.index)
      } 
    />;
  }

  render() {
    return (
     <div className='chord-diagram'>
        <div className='title'>{this.props.title}</div>
        <div className='fretboard'>
          {this.renderStrings()}
          {this.renderFrets()}
          {this.renderDots()}
          {this.renderDoubleDots()}
        </div>
        {this.renderDeleteButton()}
     </div>
    )
  }
}

export default connect(
  undefined,
  dispatch => { 
    return { 
      removeChordDiagram: (index) => {
        return dispatch(removeChordDiagram(index));
      }
    }; 
  }
)(ChordDiagram);
