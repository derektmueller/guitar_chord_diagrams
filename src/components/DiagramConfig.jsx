import React from 'react';
import { setChordDiagrams } from '../actions/chordDiagrams'
import { connect } from 'react-redux'

export class DiagramConfig extends React.Component {
  constructor(props) {
    super();

    this.state = {
      config: JSON.stringify(props.chordDiagrams)
    }
  }

  componentWillReceiveProps(nextProps) {
    this.setState({config: JSON.stringify(nextProps.chordDiagrams)});
  }

  setChordDiagrams() {
    this.props.setChordDiagrams(this.state.config);
  }

  updateConfig(event) {
    this.setState({config: event.target.value}); 
  }

  render() {
    return (<div className='diagram-config'>
      <textarea 
        onChange={this.updateConfig.bind(this)}
        value={this.state.config} 
      />
      <button type='submit' 
        onClick={this.setChordDiagrams.bind(this)}>
        Update Diagrams</button>
    </div>);
  }
};

export default connect(
  state => { return { chordDiagrams: state.chordDiagrams }; },
  dispatch => { 
    return { 
      setChordDiagrams: (index) => {
        return dispatch(setChordDiagrams(index));
      }
    }; 
  }
)(DiagramConfig);
