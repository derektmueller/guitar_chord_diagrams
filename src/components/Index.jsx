import React from 'react';
import ChordDiagram from "./ChordDiagram.jsx";
import ChordConfig from "./ChordConfig.jsx";
import ChordDiagrams from '../../lib/ChordDiagrams.js';
import { addChordDiagram } from '../actions/chordDiagrams'
import { connect } from 'react-redux'

class ChordDiagramModel {
  constructor({fretCount, notes, tuning, title, root}) {
    let colorPalette = {
      e: 'BB453C',
      g: '7D9F13',
      a: 'E7DB84',
      b: 'A2D2E9',
      d: '3EB88D'
    };
    Object.assign(this, {
      chordDiagrams: new ChordDiagrams({
        fretCount: parseInt(fretCount, 10), notes, tuning
      }),
      title,
      colorPalette,
      root
    });
  }
}

export class Index extends React.Component {
  constructor(props) {
    super(props);

//    this.props.addChordDiagram({ 
//        fretCount: 13, 
//        notes: 'e g a b d',
//        title: "e minor pentatonic",
//        root: 'e'
//    });
//    this.props.addChordDiagram({ 
//        fretCount: 13, 
//        notes: 'e g b',
//        title: "e minor (guitar)",
//        root: 'e'
//    });
//    this.props.addChordDiagram({ 
//        fretCount: 13, 
//        notes: 'e g b',
//        title: "e minor (ukulele)",
//        tuning: 'g c e a',
//        root: 'e'
//    });
  }

  renderDiagrams() {
    return this.props.chordDiagrams.map((chordDiagram, i) => {
      return <ChordDiagram 
        {...new ChordDiagramModel(chordDiagram)}
        key={i}
      />;
    });
  }

  render() {
    return (
      <div id='app'>
        <ChordConfig />
        <div className='diagrams'>
        {this.renderDiagrams()}
      </div>
    </div>);
  }
}

export default connect(
  state => { return { chordDiagrams: state.chordDiagrams }; },
  dispatch => { 
    return { 
      addChordDiagram: (config) => dispatch(addChordDiagram(config))
    }; 
  }
)(Index);
