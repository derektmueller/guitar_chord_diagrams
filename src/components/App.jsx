import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { createStore, combineReducers, applyMiddleware } from 'redux'

import reducers from '../reducers';
import UrlState from '../middleware/urlState';
import Index from "./Index.jsx";
const urlState = new UrlState(['chordDiagrams']);

const store = createStore(
  reducers,
  urlState.getPreloadedState(),
  applyMiddleware(urlState.getMiddleware()));

export default class App extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <Index />
      </Provider>
    );
  }
}
