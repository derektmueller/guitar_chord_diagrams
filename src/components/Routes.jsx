import React from 'react';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
import Editor from "./Editor.jsx";
import Index from "./Index.jsx";

export default class Routes extends React.Component {
  render() {
    return <Router>
      <div>
        <nav>
          <Link to='/'>Fretboard Diagrams</Link>
          <Link to='/' className='home'>Home</Link>
          <Link to='/editor' className='editor'>Editor</Link>
        </nav>
        <Route path="/" exact component={Index} />
        <Route path="/editor" component={Editor} />
      </div>
    </Router>;
  }
}
