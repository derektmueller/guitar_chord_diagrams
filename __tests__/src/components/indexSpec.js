import React from 'react';
import renderer from 'react-test-renderer';

import { Index } from '../../../src/components/Index';

jest.mock('../../../src/components/ChordConfig');

describe("creating chord diagrams", () => {
  it('renders correctly', () => {
    let index = renderer.create(<Index 
      addChordDiagram={() => {}}
      chordDiagrams={[
        { 
          fretCount: 13, 
          notes: 'e g a b d',
          title: "e minor pentatonic",
          root: 'e'
        },
        { 
          fretCount: 13, 
          notes: 'e g b',
          title: "e minor (guitar)",
          root: 'e'
        },
        { 
          fretCount: 13, 
          notes: 'e g b',
          title: "e minor (ukulele)",
          tuning: 'g c e a',
          root: 'e'
        }
      ]}
    />);
    expect(index.toJSON()).toMatchSnapshot();
  });
});
