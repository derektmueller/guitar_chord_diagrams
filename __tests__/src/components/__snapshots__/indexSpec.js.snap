// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating chord diagrams renders correctly 1`] = `
<div
  id="app"
>
  <div
    className="diagrams"
  >
    <div
      className="chord-diagram"
    >
      <div
        className="title"
      >
        e minor pentatonic
      </div>
      <div
        className="fretboard"
      >
        <div
          className="string-container"
          style={
            Object {
              "height": "16.666666666666664%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "0%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "23.076923076923077%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#E7DB84",
                "left": "38.46153846153846%",
              }
            }
          >
            <div>
              a
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "53.84615384615385%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#3EB88D",
                "left": "76.92307692307692%",
              }
            }
          >
            <div>
              d
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "92.3076923076923%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
        </div>
        <div
          className="string-container"
          style={
            Object {
              "height": "16.666666666666664%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "0%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#3EB88D",
                "left": "23.076923076923077%",
              }
            }
          >
            <div>
              d
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "38.46153846153846%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "61.53846153846154%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#E7DB84",
                "left": "76.92307692307692%",
              }
            }
          >
            <div>
              a
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "92.3076923076923%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
        </div>
        <div
          className="string-container"
          style={
            Object {
              "height": "16.666666666666664%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "0%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#E7DB84",
                "left": "15.384615384615385%",
              }
            }
          >
            <div>
              a
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "30.76923076923077%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#3EB88D",
                "left": "53.84615384615385%",
              }
            }
          >
            <div>
              d
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "69.23076923076923%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "92.3076923076923%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
        </div>
        <div
          className="string-container"
          style={
            Object {
              "height": "16.666666666666664%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note "
            style={
              Object {
                "background": "#3EB88D",
                "left": "0%",
              }
            }
          >
            <div>
              d
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "15.384615384615385%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "38.46153846153846%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#E7DB84",
                "left": "53.84615384615385%",
              }
            }
          >
            <div>
              a
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "69.23076923076923%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#3EB88D",
                "left": "92.3076923076923%",
              }
            }
          >
            <div>
              d
            </div>
          </div>
        </div>
        <div
          className="string-container"
          style={
            Object {
              "height": "16.666666666666664%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note "
            style={
              Object {
                "background": "#E7DB84",
                "left": "0%",
              }
            }
          >
            <div>
              a
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "15.384615384615385%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#3EB88D",
                "left": "38.46153846153846%",
              }
            }
          >
            <div>
              d
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "53.84615384615385%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "76.92307692307692%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#E7DB84",
                "left": "92.3076923076923%",
              }
            }
          >
            <div>
              a
            </div>
          </div>
        </div>
        <div
          className="string-container"
          style={
            Object {
              "height": "16.666666666666664%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "0%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "23.076923076923077%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#E7DB84",
                "left": "38.46153846153846%",
              }
            }
          >
            <div>
              a
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "53.84615384615385%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#3EB88D",
                "left": "76.92307692307692%",
              }
            }
          >
            <div>
              d
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "92.3076923076923%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
        </div>
        <div
          className="frets"
        >
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
        </div>
        <div
          className="dot"
          style={
            Object {
              "left": "19.23076923076923%",
            }
          }
        />
        <div
          className="dot"
          style={
            Object {
              "left": "34.61538461538461%",
            }
          }
        />
        <div
          className="dot"
          style={
            Object {
              "left": "50%",
            }
          }
        />
        <div
          className="dot"
          style={
            Object {
              "left": "65.38461538461539%",
            }
          }
        />
        <div
          className="double-dots"
        >
          <div
            className="dot"
            style={
              Object {
                "left": "88.46153846153847%",
              }
            }
          />
          <div
            className="dot"
            style={
              Object {
                "left": "88.46153846153847%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="chord-diagram"
    >
      <div
        className="title"
      >
        e minor (guitar)
      </div>
      <div
        className="fretboard"
      >
        <div
          className="string-container"
          style={
            Object {
              "height": "16.666666666666664%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "0%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "23.076923076923077%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "53.84615384615385%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "92.3076923076923%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
        </div>
        <div
          className="string-container"
          style={
            Object {
              "height": "16.666666666666664%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "0%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "38.46153846153846%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "61.53846153846154%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "92.3076923076923%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
        </div>
        <div
          className="string-container"
          style={
            Object {
              "height": "16.666666666666664%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "0%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "30.76923076923077%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "69.23076923076923%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "92.3076923076923%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
        </div>
        <div
          className="string-container"
          style={
            Object {
              "height": "16.666666666666664%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "15.384615384615385%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "38.46153846153846%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "69.23076923076923%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
        </div>
        <div
          className="string-container"
          style={
            Object {
              "height": "16.666666666666664%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "15.384615384615385%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "53.84615384615385%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "76.92307692307692%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
        </div>
        <div
          className="string-container"
          style={
            Object {
              "height": "16.666666666666664%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "0%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "23.076923076923077%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "53.84615384615385%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "92.3076923076923%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
        </div>
        <div
          className="frets"
        >
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
        </div>
        <div
          className="dot"
          style={
            Object {
              "left": "19.23076923076923%",
            }
          }
        />
        <div
          className="dot"
          style={
            Object {
              "left": "34.61538461538461%",
            }
          }
        />
        <div
          className="dot"
          style={
            Object {
              "left": "50%",
            }
          }
        />
        <div
          className="dot"
          style={
            Object {
              "left": "65.38461538461539%",
            }
          }
        />
        <div
          className="double-dots"
        >
          <div
            className="dot"
            style={
              Object {
                "left": "88.46153846153847%",
              }
            }
          />
          <div
            className="dot"
            style={
              Object {
                "left": "88.46153846153847%",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="chord-diagram"
    >
      <div
        className="title"
      >
        e minor (ukulele)
      </div>
      <div
        className="fretboard"
      >
        <div
          className="string-container"
          style={
            Object {
              "height": "25%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "15.384615384615385%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "53.84615384615385%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "76.92307692307692%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
        </div>
        <div
          className="string-container"
          style={
            Object {
              "height": "25%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "0%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "23.076923076923077%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "53.84615384615385%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "92.3076923076923%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
        </div>
        <div
          className="string-container"
          style={
            Object {
              "height": "25%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "30.76923076923077%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "53.84615384615385%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "84.61538461538461%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
        </div>
        <div
          className="string-container"
          style={
            Object {
              "height": "25%",
            }
          }
        >
          <div
            className="string"
          />
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "0%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#A2D2E9",
                "left": "30.76923076923077%",
              }
            }
          >
            <div>
              b
            </div>
          </div>
          <div
            className="note  root"
            style={
              Object {
                "background": "#BB453C",
                "left": "69.23076923076923%",
              }
            }
          >
            <div>
              e
            </div>
          </div>
          <div
            className="note "
            style={
              Object {
                "background": "#7D9F13",
                "left": "92.3076923076923%",
              }
            }
          >
            <div>
              g
            </div>
          </div>
        </div>
        <div
          className="frets"
        >
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
          <div
            className="fret"
            style={
              Object {
                "width": "7.6923076923076925%",
              }
            }
          />
        </div>
        <div
          className="dot"
          style={
            Object {
              "left": "19.23076923076923%",
            }
          }
        />
        <div
          className="dot"
          style={
            Object {
              "left": "34.61538461538461%",
            }
          }
        />
        <div
          className="dot"
          style={
            Object {
              "left": "50%",
            }
          }
        />
        <div
          className="dot"
          style={
            Object {
              "left": "65.38461538461539%",
            }
          }
        />
        <div
          className="double-dots"
        >
          <div
            className="dot"
            style={
              Object {
                "left": "88.46153846153847%",
              }
            }
          />
          <div
            className="dot"
            style={
              Object {
                "left": "88.46153846153847%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
